arches/app/datatypes/core/geojson_feature_collection.py

```diff
@@ -0,0 +1,430 @@
+                            value,
+                            source,
+                            message,
+                            "This data was not imported.",
+                        ),
+                        "title": title,
+                    }
+                )
+
+        if value is not None:
+            for feature in value["features"]:
+                try:
+                    geom = GEOSGeometry(JSONSerializer().serialize(feature["geometry"]))
+                    if geom.valid:
+                        validate_geom_bbox(geom)
+                    else:
+                        raise Exception
+                except Exception:
+                    message = _("Unable to serialize some geometry features.")
+                    title = _("Unable to Serialize Geometry")
+                    error_message = self.create_error_message(
+                        value, source, row_number, message, title
+                    )
+                    errors.append(error_message)
+        return errors
+
+    def to_json(self, tile, node):
+        data = self.get_tile_data(tile)
+        if data:
+            return self.compile_json(tile, node, geojson=data.get(str(node.nodeid)))
+
+    def clean(self, tile, nodeid):
+        if tile.data[nodeid] is not None and "features" in tile.data[nodeid]:
+            if len(tile.data[nodeid]["features"]) == 0:
+                tile.data[nodeid] = None
+
+    def check_geojson_value(self, value):
+        if type(value) is str:
+            geojson = json.loads(value)
+        else:
+            geojson = value
+
+        features = []
+        if geojson["type"] == "FeatureCollection":
+            for feature in geojson["features"]:
+                if feature["geometry"]["type"].find("Multi") != -1:
+                    new_collection = self.geo_utils.convert_multipart_to_singlepart(
+                        feature["geometry"]
+                    )
+                    for new_feature in new_collection["features"]:
+                        new_feature["id"] = (
+                            geojson["id"] if "id" in geojson else str(uuid.uuid4())
+                        )
+                    features = features + new_collection["features"]
+                else:
+                    # keep the feature id if it exists, or generate a fresh one.
+                    feature["id"] = (
+                        feature["id"] if "id" in feature else str(uuid.uuid4())
+                    )
+                    features.append(feature)
+            geojson["features"] = features
+            return geojson
+        else:
+            raise TypeError
+
+    def transform_value_for_tile(self, value, **kwargs):
+        if "format" in kwargs and kwargs["format"] == "esrijson":
+            arches_geojson = self.geo_utils.arcgisjson_to_geojson(value)
+        else:
+            try:
+                arches_geojson = self.check_geojson_value(value)
```

Comment:
This check also needs to apply to the arches_json on line 134 in case multiparts are coming as `arcgisjson` (as would be the case from the esri addin).

Reply:
a bit circuitous with the dict conversion, but this should work.